src/components/shared/Entity/Actions/SchemaEvolution.tsx

```diff
@@ -0,0 +1,169 @@
+
+interface Props {
+    onFailure: (formState: Partial<FormState>) => void;
+}
+
+function SchemaEvolution({ onFailure }: Props) {
+    const supabaseClient = useClient();
+    const storeDiscoveredCollections = useStoreDiscoveredCaptures();
+
+    const entityType = useEntityType();
+    const editingEntity = useEntityWorkflow_Editing();
+
+    // Draft Editor Store
+    const draftId = useEditorStore_id();
+    const isSaving = useEditorStore_isSaving();
+
+    // Form State Store
+    const setFormState = useFormStateStore_setFormState();
+    const updateFormStatus = useFormStateStore_updateStatus();
+    const formActive = useFormStateStore_isActive();
+
+    const incompatibleCollections =
+        useBindingsEditorStore_incompatibleCollections();
+
+    const setIncompatibleCollections =
+        useBindingsEditorStore_setIncompatibleCollections();
+
+    const jobFailed = useCallback(
+        (error) => {
+            setFormState({
+                error,
+                status: FormStatus.FAILED,
+            });
+        },
+        [setFormState]
+    );
+
+    const waitForEvolutionToFinish = (
+        logTokenVal: string,
+        draftIdVal: string
+    ) => {
+        updateFormStatus(FormStatus.SCHEMA_EVOLVING);
+
+        jobStatusPoller(
+            supabaseClient
+                .from(TABLES.EVOLUTIONS)
+                .select(`${JOB_STATUS_COLUMNS}, collections, draft_id`)
+                .match({
+                    draft_id: draftIdVal,
+                    logs_token: logTokenVal,
+                }),
+            async (payload: any) => {
+                // Update the state so it knows that things came back correctly
+                setFormState({
+                    status: FormStatus.SCHEMA_EVOLVED,
+                });
+
+                // Treat these like discovered collections. This will mean the UI
+                //  will go fetch all the new collections, specs, configs and store
+                //  them locally and present them
+                await storeDiscoveredCollections(
+                    payload.draft_id,
+                    entityType,
+                    onFailure
+                );
+
+                // Now that the collections are updated we can clear out the list so
+                //  the error section is no longer displayed.
+                setIncompatibleCollections([]);
+            },
+            async (payload: any) => {
+                if (payload.error === JOB_STATUS_POLLER_ERROR) {
+                    jobFailed(DEFAULT_POLLER_ERROR);
+                } else {
+                    jobFailed({
+                        ...DEFAULT_POLLER_ERROR,
+                        message: `entityEvolution.serverUnreachable`,
+                    });
+                }
+            }
+        );
+    };
+
+    const save = async (event: React.MouseEvent<HTMLElement>) => {
+        event.preventDefault();
+
+        updateFormStatus(FormStatus.SCHEMA_EVOLVING);
```

Comment:
Added a new status to make things a bit more clear even though we do not trigger much off the status